docs/specs/buffer.md

```diff
@@ -0,0 +1,50 @@
+# Buffer Specification
+
+This document specifies Vector's buffer behavior for the development of Vector.
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”,
+“SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be
+interpreted as described in [RFC 2119].
+
+<!-- MarkdownTOC autolink="true" style="ordered" indent="   " -->
+
+1. [Instrumentation](#instrumentation)
+
+<!-- /MarkdownTOC -->
+
+## Instrumentation
+
+Vector buffers MUST be instrumented for optimal observability and monitoring. This is required to drive various interfaces that Vector users depend on to manage Vector installations in mission critical production environments. This section extends the [Instrumentation Specification].
+
+### Events
+#### `EventsReceived`
+*All buffers* MUST emit an `EventsReceived` event immediately after receiving one or more Vector events.
+* Properties
+  * `entity` - buffer
+  * `count` - the number of received events
+  * `byte_size` - the byte size of received events
+* Metric
+  * MUST increment the `received_events_total` counter by the defined `count` with other properties as tags
```

Comment:
Can you add a sub-list with any relevant tags? Specifically the tag that would differentiate this from components?

Reply:
This also depends on what we do for https://github.com/vectordotdev/vector/pull/9165/files#r711134056

Reply:
Sure thing, can add an `entity: buffer` note.